### ISSUE TITLE
SSH Remote development not working on latest Darwin ARM64 Experimental build

### ISSUE THREAD

**Issue by rukkora** [microsoft/vscode#111342]
- VSCode Version:Version: 1.52.0-exploration
(Commit: f9b242d7e68aec36dc08ee7840d8f3a2b22b51d5(
- OS Version: macOS 11.0.1 on MacBook Air 2020

Steps to Reproduce:

I installed VSCode 1.51.1 (Stable) and 1.52.0-exploration. SSH Remote Development works on 1.15.1 (Stable), but it doesn't work on 1.52.0-exploration.
I tried some SSH servers including raspberry pi and x86 server, but it didn't work on both of them outputting the same error.
There's the error log below.

```
[13:36:41.540] Downloading VS Code server locally...
[13:36:41.666] Terminating local server
[13:36:41.667] Resolver error: Error: Server returned 404
    at Object.t.asText (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:1999:259)
    at l.download (file:///Applications/Visual Studio Code - Exploration.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.js:2000:818)
[13:36:41.669] Local server exit: 15
```

Does this issue occur when all extensions are disabled?: I only enabled SSH extension.

**Comment by ItalyPaleAle** [comment]
This is impacting Remote - Containers too, with this error:

```
[30306 ms] Start: Downloading VS Code Server
[30484 ms] Failed to download VS Code Server (https://update.code.visualstudio.com/commit:f9b242d7e68aec36dc08ee7840d8f3a2b22b51d5/server-linux-x64/exploration): HTTP 404 - Not Found
```

The issue is that the latest build was compiled from commit `f9b242d7e68aec36dc08ee7840d8f3a2b22b51d5` but only the Mac/ARM binaries were built and not the Linux ones, so the server bits are missing.

**Comment by deepak1556** [comment]
This should be fixed in latest release a31f124319 , please check. Thanks!
